File: app/src/main/java/com/example/elsafa/SignUp/OnSignUpListener.java
package com.example.elsafa.SignUp;

import androidx.annotation.NonNull;

import com.google.firebase.FirebaseException;
import com.google.firebase.auth.PhoneAuthCredential;
import com.google.firebase.auth.PhoneAuthProvider;

import Model.FirebaseUser;

public interface OnSignUpListener {

    void onVerificationCompleted(PhoneAuthCredential credential);

    void onVerificationFailed(FirebaseException e);

    void onCodeSent(@NonNull String verificationId,
                    @NonNull PhoneAuthProvider.ForceResendingToken token);

    void onGetUserDataSuccess(FirebaseUser firebaseUser);

    void onGetUserDataFailed(Exception e);

    void onSignInFailed();

    void onSignInSuccess();
}

File: app/src/main/java/com/whiteside/cafe/SignUp/OnSignUpListener.kt
package com.whiteside.cafe.SignUp

import Model.FirebaseUser
import com.google.firebase.FirebaseException
import com.google.firebase.auth.PhoneAuthCredential
import com.google.firebase.auth.PhoneAuthProvider.ForceResendingToken

interface OnSignUpListener {
    open fun onVerificationCompleted(credential: PhoneAuthCredential?)
    open fun onVerificationFailed(e: FirebaseException?)
    open fun onCodeSent(
        verificationId: String,
        token: ForceResendingToken
    )

    open fun onGetUserDataSuccess(firebaseUser: FirebaseUser?)
    open fun onGetUserDataFailed(e: Exception?)
    open fun onSignInFailed()
    open fun onSignInSuccess()
}